lib/engine/game/g_1866/game.rb

```diff
@@ -950,20 +980,100 @@ def upgrades_to?(from, to, _special = false, selected_company: nil)
           super
         end
 
+        def can_par_share_price?(share_price, corporation)
+          return (share_price.corporations.empty? || share_price.price == self.class::MAX_PAR_VALUE) unless corporation
+
+          share_price.corporations.none? { |c| c.type != :stock_turn_corporation } ||
+            share_price.price == self.class::MAX_PAR_VALUE
+        end
+
+        def convert_public_corporation?
```

Comment:
what's a public corporation, is there a private corporation?

Reply:
In rules there are many different types of companies, but the 2 with private and public is
private company = company in 18xx.games
public company = corporation

Ive just looked at the rules to much and copied :)